File: 0509 Fibonacci Number.py
class Solution:
    def fib(self, N: int) -> int:
        if N == 0: return 0
        if N == 1: return 1
        f0, f1 = 0, 1
        for i in range(N-1):
            f0, f1 = f1, f0 + f1
        return f1

File: 0509 Fibonacci Number.java
class Solution {
    public int fib(int N) {
        if (N == 0) return 0;
        if (N == 1) return 1;
        int f0 = 0, f1 = 1, f2 = 1;
        while (N-- > 2) {
            f0 = f1;
            f1 = f2;
            f2 = f0 + f1;
        }
        return f2;
    }
}